app/assets/javascripts/utils/api.js

```diff
@@ -179,6 +179,21 @@ const API = {
     return response.json();
   },
 
+  async cloneCourseWithAssignment(id, campaign) {
```

Comment:
Again, I think there's no need to create a new method; the old one should be able accommodate both behaviors.